app/api/regions.py

```diff
@@ -51,24 +133,40 @@ async def list_regions(
     current_user: User = Depends(get_current_user_from_auth),
     db: Session = Depends(get_db)
 ):
-    return db.query(DBRegion).filter(DBRegion.is_active == True).all()
+    # System admin users can see all regions
+    if current_user.is_admin:
+        return db.query(DBRegion).filter(DBRegion.is_active == True).all()
+
+    # Regular users can only see non-dedicated regions
+    if not current_user.team_id:
+        return db.query(DBRegion).filter(
+            DBRegion.is_active == True,
+            DBRegion.is_dedicated == False
+        ).all()
+
+    # Team members can see non-dedicated regions plus their team's dedicated regions
+    team_dedicated_regions = db.query(DBRegion).join(DBTeamRegion).filter(
+        DBRegion.is_active == True,
+        DBRegion.is_dedicated == True,
+        DBTeamRegion.team_id == current_user.team_id
+    ).all()
 
-@router.get("/admin", response_model=List[Region])
+    non_dedicated_regions = db.query(DBRegion).filter(
+        DBRegion.is_active == True,
+        DBRegion.is_dedicated == False
+    ).all()
```

Comment:
nit: you could use the same variable for the "regular user" return value above to avoid duplicating the business logic.